docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,531 @@
+
+```scala
+package x.y.z
+
+@EnableReflectiveInstantation
+sealed trait ReflectiveFoo {
+  val value: String = "foo"
+}
+
+object SingleFoo extends ReflectiveFoo
+
+case class MultipleFoo(times: Int) extends ReflectiveFoo {
+  override val value: String = super.value * times
+}
+
+for {
+  cls <- lookupInstantiatableClass("x.y.z.Foo")
+  ctor <- cls.getConstructor(classOf[Int])
+  obj <- ctor.newInstance(5)
+} yield obj // results in Some(new MultipleFoo(5))
+
+for {
+  cls <- lookupLoadableModule("x.y.z.SingleFoo")
+  obj <- cls.loadModule
+} yield obj // results Some(SingleFoo)
+```
+
+### Cross compilation
+With the latest release you're able to define a custom target for compiler by providing [LLVM style TargetTriple](https://clang.llvm.org/docs/CrossCompilation.html) in your config.
+This change does not change default behaviour of Scala Native sbt plugin, cousing to target host architecture and operating system.
+
+For example, if you're working on Linux and would like to create perfectly suited assembly for some specific version of MacOS without changing your whole build,
+you can use ad-hoc sbt setting:
+```bash
+sbt 'set nativeConfig ~= {_.withTargetTriple("x86_64-apple-darwin<version>")}' myApp/nativeLink
+```
+
+We consider changing target triple as a feature for advanced users, and cannot promise it would currently work with any possible configuration yet.
+However, the number of supported architectures and operating systems would definitely grow in the future.
+
+> When using Linux / MacOS you can check target triple used in your environment by command `llvm-config --host-target`.
+
+### Native sources in build
+With the 0.4.0 release you're able to put your C/C++ sources in the `resources/scala-native` directory inside your project,
+so they will be linked and compiled inside the SN pipeline.
+
+As an example you can use it to access macro defined constants and functions or to pass `struct`s from the stack to C functions.
+```c
+// src/resources/scala-native/example.c
+typedef int (*Callback0) (void);
+
+const int EXAMPLE_CONSTANT = 42;
+
+int exec(Callback0 f) {
+ return f();
+};
+
+```
+```scala
+// src/main/example.scala
+@extern
+object example {
+ def exec(cb: CFuncPtr0[CInt]): ExecResult = extern
+
+ @name("EXAMPLE_CONSTANT")
+ final val someConstant: Int = extern
+}
+```
+
+### Commix GC
+This release also adds a new Garbage Collector - Commix, a parallel mark, and concurrent sweep GC, based on the well known Immix GC.
+It reduces GC pause times by utilizing additional processor cores during mark and sweep phases. It was added soon after
+the 0.4.0-M2 release, but was only available via an unofficial snapshot.
+
+Remember, it **does not** mean Scala Native is starting to officially support multi-threading!
+Commix GC was written in C and uses `pthread` to work. In case if your application needs concurrency support you may try experimental library [scala-native-loop](https://github.com/scala-native/scala-native-loop)
```

Comment:
```suggestion
Commix GC was written in C and uses `pthread` to work. In case your application needs concurrency support, you may try the experimental library [scala-native-loop](https://github.com/scala-native/scala-native-loop)
```